homeassistant/components/homekit/__init__.py

```diff
@@ -480,10 +543,14 @@ def _start(self, bridged_states):
 
         if not self.driver.state.paired:
             show_setup_message(
```

Comment:
Why is this not part of the config flow? So we finish the flow when the pairing is done. We can do this with external step. I think that we might have to adjust external step to also work without a url. 

This PR is already big, so we can also postpone this for a future PR.

Reply:
Pairing can happen outside of the config flow.  If they unpair they can repair on another device.

Reply:
Couldn't we start a new pair config flow in that case if we get unpaired that just updates the existing flow?

Reply:
I've been watching https://github.com/home-assistant/architecture/issues/377 to see how it pans out as it might be a similar use case.

I'll do a followup pending the outcome of the above (it could turn out to not be relevant to this one).